modules/git/command.go

```diff
@@ -86,7 +86,8 @@ func (c *Command) RunInDirTimeoutEnvFullPipeline(env []string, timeout time.Dura
 	if env == nil {
 		cmd.Env = append(os.Environ(), fmt.Sprintf("LC_ALL=%s", DefaultLCALL))
 	} else {
-		cmd.Env = append(env, fmt.Sprintf("LC_ALL=%s", DefaultLCALL))
+		env = append(env, fmt.Sprintf("LC_ALL=%s", DefaultLCALL))
+		cmd.Env = env
```

Comment:
Sorry, this is not what I meant. I was asking whether `env` (when set) had `os.Environ()` in it or if you had to take care of it here instead. I've cloned you PR and now I see that the caller ([`repo_tree.go:(*Repository).CommitTree()`](https://github.com/zeripath/gitea/blob/18300d04c46a3d55cfa6333995cfe88ef12c63f4/modules/git/repo_tree.go#L76-L83)) is already taking care of it (I couldn't see about that from the diff alone).

So you can disregard my question. However, this new code change seems unnecessary; `cmd.Env = append....` was alright.

Reply:
golangci-lint required that I wrote it that way.

Reply:
Probably it's better as ```go
cmd.Env = env
cmd.Env = append(cmd.Env, fmt.Sprintf("LC_ALL=%s", cmd.LCALL))
```

Reply:
> 
> 
> golangci-lint required that I wrote it that way.

"If your linter is asking you to write bad code, it's probably not well configured." - _Chinese Proverb_

Reply:
It really can be a struggle